SYCL/Regression/set-arg-local-accessor.cpp

```diff
@@ -0,0 +1,79 @@
+// REQUIRES: opencl
+//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -lOpenCL -o %t.out
+// RUN: env SYCL_DEVICE_FILTER=opencl %CPU_RUN_PLACEHOLDER %t.out
+// RUN: env SYCL_DEVICE_FILTER=opencl %GPU_RUN_PLACEHOLDER %t.out
+// RUN: env SYCL_DEVICE_FILTER=opencl %ACC_RUN_PLACEHOLDER %t.out
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+const char *clSource = R"(
+    kernel void test(global int *a, local float *b, int n) {
+        if (get_local_id(0) != 0)
+            return;
+        for (int i = 0; i < n; i++)
+            b[i] = i;
+        barrier(CLK_LOCAL_MEM_FENCE);
```

Comment:
Could you please clarify why the `barrier` is required here?

Reply:
It is not needed indeed. Refactored kernel so the barrier is useful in 8cfb6f4c98e7f7d71f6d7f2da3d8bb37e0496759 and 3620717208f1d81ac6b0d29fd7c00ffbad0864b7